### PR TITLE
Positivity cuts

### PR DESCRIPTION
**Pull request by comane**
This PR addresses #2058 

Main modifications:

-  removed `load_commondata` method from LagrangeSetSpec(DataSetSpec) so that it is directly inherited from `DataSetSpec` where cuts are loaded (before no cuts were loaded for LagrangeSetSpec type datasets)
- For a `LagrangeSetSpec` promoted `Cuts` to `InternalCutsWrapper` so as to allow for the proper loading of cuts. --> note: rules need now to be passed to `DataSetSpec`
- renamed kin of XGL and added new process for that positivity dataset
- Changed the names of some INTEG and POS processes

REPORTS
----------
- compare fits: https://vp.nnpdf.science/68xE8NcIQhu2i2hAwhII5A==/Scales0_pdf_report_report.html

### CODE REVIEW

File: n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml [hunk 1]
```diff
@@ -0,0 +1,189 @@
+#
```
Comment (scarlehoff) [inline n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml:1]: Please use a different name for this runcard. We should have only one `nnpdf40-like` to avoid generating confusion again.

Also, since this is just an example you can do with a subset of data.

File: validphys2/src/validphys/core.py [hunk 2]
```diff
@@ -586,10 +594,6 @@ def to_unweighted(self):
         )
         return self
 
-    @functools.lru_cache
-    def load_commondata(self):
-        return self.commondata.load()
-
```
Comment (scarlehoff) [inline validphys2/src/validphys/core.py:592]: Why has this been removed? Is it not used anywhere anymore?

File: n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    - dataset: NNPDF_POS_2P24GEV_XDQ
+      rule: "x > 0.1"
+
+    - dataset: NNPDF_POS_2P24GEV_XDB
+      rule: "x > 0.1"
+
+    - dataset: NNPDF_POS_2P24GEV_XSQ
+      rule: "x > 0.1"
+
+    - dataset: NNPDF_POS_2P24GEV_XSB
```
Comment (scarlehoff) [inline n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml:178]: Given that you are setting the `POS_XPDF` process type, you should be able to also do 

```suggestion
    - process_type: POS_XPDF
```

and have one single cut.

**Review by scarlehoff — COMMENTED** [review]
Thanks! Only some minor comments. Could you also link in this PR some of the fits you've done using this branch?

File: nnpdf_data/nnpdf_data/new_commondata/NNPDF_POS_2P24GEV/metadata.yaml [hunk 4]
```diff
@@ -56,7 +56,7 @@ implemented_observables:
     description: Deep Inelastic Scattering
     label: 'positivity dataset: DIS $c+\bar{c}$ structure function $F_2^c$'
     units: ''
-  process_type: DIS_F2C
+  process_type: POS_F2C
```
Comment (RoyStegeman) [inline nnpdf_data/nnpdf_data/new_commondata/NNPDF_POS_2P24GEV/metadata.yaml:59]: Why this change? The process is DIS_F2C, the only difference is that we use the dataset in a lagrange multiplier. Also, this change is not mirrored in `process_options.py`

File: n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml (hunk 1 above)
Comment (comane) [reply n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml:1]: Yes but at the same time the `Basic_runcard.yml` does not have any MS-bar POS sets.
So I would rather have something called:
`Basic_runcard_ms_bar_pos_cuts.yaml`, where the runcard is the same as the current nnpdf40-like_pos_cut.yaml apart from reduced datasets.
What do you think?

File: validphys2/src/validphys/core.py (hunk 2 above)
Comment (comane) [reply validphys2/src/validphys/core.py:592]: Yes exactly, this shouldn't be used anywhere else.

File: n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml (hunk 1 above)
Comment (scarlehoff) [reply n3fit/runcards/examples/nnpdf40-like_pos_cut.yaml:1]: You can add a MS-bar POS set to the `Basic_runcard.yml`

File: n3fit/runcards/examples/Basic_runcard.yml [hunk 5]
```diff
@@ -73,12 +73,17 @@ positivity:
...
   - {dataset: NNPDF_POS_2P24GEV_F2U, maxlambda: 1e6}        # Positivity Lagrange Multiplier
   - {dataset: NNPDF_POS_2P24GEV_FLL-19PTS, maxlambda: 1e6}
+  - {dataset: NNPDF_POS_2P24GEV_XGL, maxlambda: 1e6}
 
 ############################################################
 integrability:
   integdatasets:
   - {dataset: NNPDF_INTEG_3GEV_XT3, maxlambda: 1e2}
 
+added_filter_rules:
```
Comment (scarlehoff) [inline n3fit/runcards/examples/Basic_runcard.yml:83]:
```suggestion
# Apply a cut to a dataset or process type, in this case to the positivity dataset `NNPDF_POS_2P24GEV_XGL` which has process type: POS_XPDF
added_filter_rules:
```

File: nnpdf_data/nnpdf_data/new_commondata/NNPDF_POS_2P24GEV/metadata.yaml (hunk 4 above)
Comment (scarlehoff) [reply nnpdf_data/nnpdf_data/new_commondata/NNPDF_POS_2P24GEV/metadata.yaml:59]: I think it is better to keep it as a `POS_F2C`. The positivity are special. I would however call them all `POS_F2`?
The other reason why I think they should not be called `DIS_` is that there's a lot of hidden logic that treats `DIS_` processes differently, better  to keep the positivity ones differentiated.

Agree that a `POS_F2` would be needed in process type (I guess you are already doing it like that but positivity/integrability datasets don't need kinematic coverage so no need to implement one).

**Review by scarlehoff — APPROVED** [review]
Left some small comments, but other than that I think this is good to go.
Make sure you rebase the latests changes in master just to make sure nothing breaks but looks good to me.

File: nnpdf_data/nnpdf_data/new_commondata/NNPDF_POS_2P24GEV/metadata.yaml (hunk 4 above)
Comment (comane) [reply nnpdf_data/nnpdf_data/new_commondata/NNPDF_POS_2P24GEV/metadata.yaml:59]: Just to be sure, would you change all process types starting with `POS_F2` with `POS_F2` (eg. `POS_F2C_CCP ` -> `POS_F2`)

What about other processes such as `POS_FLL` should we just leave them as they are?

Comment (comane) [reply nnpdf_data/nnpdf_data/new_commondata/NNPDF_POS_2P24GEV/metadata.yaml]: I did rename all of the `POS_F2*` process types to `POS_F2`. I have also added the corresponding process types in `process_options.py`
